src/utils/slack.js

```diff
@@ -0,0 +1,101 @@
+const { cache } = require("./cache");
+
+let defaultClient = {};
+
+const setClient = (client) => {
+  defaultClient = client;
+};
+
+const addEmojiReaction = async (msg, reaction) => {
+  const {
+    client,
+    event: { channel, ts: timestamp },
+  } = msg;
+
+  return client.reactions.add({ name: reaction, channel, timestamp });
+};
+
+const getChannelID = (() => {
+  const channelIDs = new Map();
+
+  return async (channelName) => {
+    if (!channelIDs.get(channelName)) {
+      const all = await defaultClient.conversations.list({
```

Comment:
This should definitely be cached. It will change every time a new channel is created, but given how the bot uses it, it won't ever _meaningfully_ changed.

Reply:
It turns out this doesn't need to be cached. It will only fetch the list of channels from Slack if the one it's looking for isn't already there, which should only happen once after load, because this function caches off a mapping of channel names to IDs itself.